modules/client/src/main/java/org/apache/ignite/internal/client/ReliableChannel.java

```diff
@@ -929,4 +931,18 @@ private boolean isFutureInProgressOrDoneAndChannelOpen(@Nullable CompletableFutu
             return ch != null && !ch.closed();
         }
     }
+
+    private void logFailedEstablishConnection(ClientChannelHolder ch, Throwable err) {
+        String logMessage = "Failed to establish connection to " + ch.chCfg.getAddress() + ": " + err.getMessage();
+
+        if (log.isDebugEnabled() || isLogFailedEstablishConnectionException(err)) {
+            log.warn(logMessage, err);
```

Comment:
Why the log level suddenly turns `WARN` when debug is enabled?

Reply:
fix it